src/main/java/seedu/address/logic/parser/FindOrderCommandParser.java

```diff
@@ -0,0 +1,66 @@
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_BLANK;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.FindOrderCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.order.Order;
+import seedu.address.model.order.OrderContainsKeywordsPredicate;
+import seedu.address.model.order.OrderPhoneContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindOrderCommand object
+ */
+public class FindOrderCommandParser implements Parser<FindOrderCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindOrderCommand
+     * and returns a FindOrderCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindOrderCommand parse(String args) throws ParseException {
+        if (args.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindOrderCommand.MESSAGE_USAGE));
+        }
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE);
+
+        Prefix findOrderPrefix;
+        Predicate<Order> findOrderPredicate;
+        if (isOnlyOnePrefixPresent(argMultimap, PREFIX_NAME)) {
+            findOrderPrefix = PREFIX_NAME;
+            String[] nameKeywords = argMultimap.getValue(findOrderPrefix).get().trim().split("\\s+");
+            findOrderPredicate = new OrderContainsKeywordsPredicate(Arrays.asList(nameKeywords));
+            return new FindOrderCommand(findOrderPredicate);
+        } else if (isOnlyOnePrefixPresent(argMultimap, PREFIX_PHONE)) {
+            findOrderPrefix = PREFIX_PHONE;
+            String[] phoneKeywords = argMultimap.getValue(findOrderPrefix).get().trim().split("\\s+");
+            findOrderPredicate = new OrderPhoneContainsKeywordsPredicate(Arrays.asList(phoneKeywords));
+            return new FindOrderCommand(findOrderPredicate);
+        } else {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindOrderCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private static boolean isOnlyOnePrefixPresent(ArgumentMultimap argumentMultimap, Prefix searchPrefix) {
+        ArrayList<Prefix> argPrefixes = argumentMultimap.getAllPrefixes();
+        // check that prefix exists
+        boolean isSearchPrefixExists = argPrefixes.stream().filter(prefix -> prefix.equals(searchPrefix)).count() == 1;
+
+        // check no other prefix exists except for blank prefix as by-product of tokenizing
+        boolean noOtherPrefixExists = argPrefixes.stream()
```

Comment:
Does this solve the issue of having extra prefixes that just gets ignored? (as mentioned in the pr)

Reply:
**Edge case in question:** `findo n/Emily d/1xchocolate cake`

**Behaviour**: Command will return search results for Orders with name "Emily"

---

This behaviour is due to how `ArgumentTokenizer.tokenize(...)` works, where the current code is now `ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE)`. With the edge case, the tokenizer will tokenize only `n/` but not tokenize `d/`. In order to tokenize `d/` the function will have to be changed to `ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_DETAIL)` instead.

From my perspective, this could pose a potential problem maintenance wise - adding a new Prefix in the future means that an update is necessary to FindOrderCommand as well, which leads to additional coupling. 

And hence I see 2 approaches to this:

1. As is the current implementation, extra attributes that are not supported for search through `findo` are ignored. However, attributes that not supported through `findo` cannot be used in tandem ie. `findo n/Emily p/93214422`.

2. Create a separate tokenizer to parse all `*/` prefixes and to include all staticPrefixes variables in `CliSyntax`.

I went with approach 1 because the help message will alert users that only filtering via `name` and `phone` is possible. Hence, any other attribute tags passed in should be superfluous. Approach 2 may have its utility, but before starting on working on it, I would want to gather feedback on whether it has usage outside of this issue as I don't see too much benefits being reaped just for this issue alone.

What do you think about this? Do you think there would be other solutions that could help better address this issue?

Reply:
I think your current implementation works well enough for our use case. I misunderstood the issue about the edge case but played around with the app and i understand what you mean now. Sorry for the miscomm.